js/app.js

```diff
@@ -0,0 +1,28 @@
+var posts = [];
+
+function Post (obj){
+  this.title = obj.title;
+  this.date = obj.date;
+  this.description = obj.description;
+  this.link = obj.link;
+}
+
+Post.prototype.toHtml = function() {
+  var $newPortfolioPiece = $('article.template').clone();
```

Comment:
Check your HTML. Do you have an article element with the class template anywhere? If you don't have that, there's nothing to clone, and this method won't work.